funcx_sdk/funcx/sdk/executor.py

```diff
@@ -170,13 +184,19 @@ def __init__(
         self._result_watcher: _ResultWatcher | None = None
 
         log.debug("%s: initiated on thread: %s", self, threading.get_ident())
-        self._task_submitter = threading.Thread(target=self._task_submitter_impl)
+        self._task_submitter = threading.Thread(
+            target=self._task_submitter_impl, name="TaskSubmitter"
+        )
         self._task_submitter.start()
+        _REGISTERED_FXEXECUTORS[id(self)] = self
 
     def __repr__(self) -> str:
         name = self.__class__.__name__
         label = self.label and f"{self.label}, " or ""
-        return f"{name}<{label}{self.batch_size}, ep_id:{self.endpoint_id}>"
+        ep_id = self.endpoint_id and f", ep_id:{self.endpoint_id}" or ""
+        c_id = self.container_id and f", c_id:{self.container_id}" or ""
```

Comment:
I'm not sure how useful it is to have the `container_id` reported, but it doesn't hurt to have it.